src/commands/org/list.ts

```diff
@@ -112,108 +141,96 @@ export class OrgListCommand extends SfCommand<OrgListResult> {
-          username: { header: 'USERNAME' },
-          orgId: { header: 'ORG ID' },
-          ...(this.flags.all || this.flags.verbose ? { status: { header: 'STATUS' } } : {}),
-          ...(this.flags.verbose
-            ? {
-                devHubOrgId: { header: 'DEV HUB' },
-                createdDate: { header: 'CREATED DATE' },
-                instanceUrl: { header: 'INSTANCE URL' },
-              }
-            : {}),
-          expirationDate: { header: 'EXPIRATION DATE' },
+      ...regularOrgs
+        .map(colorEveryFieldButConnectedStatus(chalk.magentaBright))
+        .map((row) => getStyledObject(row))
+        .map(statusToEmoji),
+
+      ...sandboxes
+        .map(addType('Sandbox'))
+        .map(colorEveryFieldButConnectedStatus(chalk.yellowBright))
+        .map((row) => getStyledObject(row))
+        .map(statusToEmoji),
+
+      ...scratchOrgs
+        .map((row) => ({ ...row, type: 'Scratch' }))
+        .map(convertScratchOrgStatus)
+        .map((row) => getStyledObject(row))
+        .map(statusToEmoji),
+    ];
+
+    this.table(
+      allOrgs.map((org) => Object.fromEntries(Object.entries(org).filter(fieldFilter))),
+      {
+        defaultMarker: {
+          header: '',
         },
-        {
-          title: 'Scratch orgs',
-        }
-      );
-    }
+        type: {
+          header: 'Type',
+        },
+        alias: {
+          header: 'Alias',
+        },
+        username: { header: 'Username' },
+        orgId: { header: 'Org ID' },
+        ...(!skipconnectionstatus ? { connectedStatus: { header: 'Status' } } : {}),
+        ...(this.flags.verbose
+          ? {
+              instanceUrl: { header: 'Instance URL' },
+              devHubOrgId: { header: 'Dev Hub ID' },
+              createdDate: {
+                header: 'Created',
+                get: (data): string => (data.createdDate as string)?.split('T')?.[0] ?? '',
+              },
+            }
+          : {}),
+        expirationDate: { header: 'Expires' },
+      }
+    );
   }
 }
 
 const decorateWithDefaultStatus = <T extends ExtendedAuthFields | FullyPopulatedScratchOrgFields>(val: T): T => ({
   ...val,
   ...(val.isDefaultDevHubUsername ? { defaultMarker: '(D)' } : {}),
   ...(val.isDefaultUsername ? { defaultMarker: '(U)' } : {}),
+  ...(val.isDefaultDevHubUsername && val.isDefaultUsername ? { defaultMarker: '(D),(U)' } : {}),
+});
+
+const statusToEmoji = <T extends ExtendedAuthFields | FullyPopulatedScratchOrgFields>(val: T): T => ({
+  ...val,
+  defaultMarker: val.defaultMarker?.replace('(D)', defaultHubEmoji)?.replace('(U)', defaultOrgEmoji),
 });
 
 // sort by alias then username
 const comparator = <T extends ExtendedAuthFields | FullyPopulatedScratchOrgFields>(a: T, b: T): number => {
-  const aliasCompareResult = (a.alias ?? '').localeCompare(b.alias ?? '');
-  return aliasCompareResult !== 0 ? aliasCompareResult : (a.username ?? '').localeCompare(b.username);
+  const emptiesLast = Array(10).fill('z').join('');
```

Comment:
NIT: lots of iterations just to fill an array with 10 'z's.. could hard code this and save _some_ time... but that's why it's a NIT 😄

Reply:
yeah